modules/metalsmith/doc_assets.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+/* eslint-env es6 */
+
+
+const {info, error} = require('./../debugger')('nx_assets');
+const fs = require('fs');
+const crypto = require('crypto');
+const Nuxeo = require('nuxeo');
+const cheerio = require('cheerio');
+
+const map = {
+    // 'a'     : 'href',
+    // 'link'  : 'href',
+    // 'script': 'src',
+    'img'  : 'src',
+    'video': 'src',
+    'audio': 'src'
+    // 'source': 'src'
+};
+
+
+const checksum = function (str, algorithm, encoding) {
+    return crypto
+        .createHash(algorithm || 'md5')
+        .update(str, 'utf8')
+        .digest(encoding || 'hex');
+};
+
+const get_attribute = function getAttribute (el) {
+    return (typeof map[el.name] !== 'undefined')
+        ? map[el.name]
+        : false;
+};
+
+const is_html = function isHTML (file) {
+    return /\.html?/.test(file);
+};
+
+
+const check_file = (files, filename, selector, nuxeo) => {
+    return new Promise(function (resolve) {
+
+        const file = files[filename];
+
+        const contents = file.contents.toString();
+        const $ = cheerio.load(contents);
+
+        const url_promises = [];
+
+        $(selector).each(function (i, el) {
+
+            const $el = $(el);
+            const attr = get_attribute(el);
+            const url = $el.attr(attr);
+            if (url && url.search(/^nx_asset?\:\/\//) !== -1) {
+                const uid = url.replace(/(^\w+:|^)\/\//, '');
+                const p = nuxeo.repository()
+                    .fetch(uid, {schemas: ['dublincore', 'file', 'document_asset']})
+                    .then(function (doc) {
+
+                        // if there is no file return
+                        if (!doc.properties['file:content']) {
+                            error('nx_asset without file:content');
+                            return null;
+                        }
+
+                        // build the right url
+                        const folder = 'nx_assets';
+                        const ext = /(?:\.([^.]+))?$/.exec(doc.properties['file:content'].name)[1];
+                        const asset_type = doc.properties['doc_asset:nature'];
+                        const asset_file = folder + '/' + uid + '+' + asset_type + '.' + ext;
+                        const href = '/' + asset_file;
+
+                        $el.attr(attr, href);
+                        const html = $.html();
+                        file.contents = new Buffer(html);
+                        files[filename] = file;
+
+                        // if file already exist in fs
+                        if (fs.existsSync(asset_file)) {
```

Comment:
This should be using the async version and resolving the promise with the file to make it transparent to the processing logic